pkg/virt-handler/rest/console.go

```diff
@@ -153,10 +153,47 @@ func (t *ConsoleHandler) VNCHandler(request *restful.Request, response *restful.
 	}
 	uid := vmi.GetUID()
 	stopChn := newStopChan(uid, t.vncLock, t.vncStopChans)
-	defer deleteStopChan(uid, stopChn, t.vncLock, t.vncStopChans)
+
+	t.handleConcurrentConnectionStart(vmi, stopChn)
+	defer t.handleConcurrentConnectionEnd(vmi, stopChn)
+
 	t.stream(vmi, request, response, unixSocketDialer(vmi, unixSocketPath), stopChn)
 }
 
+
+var vncMap sync.Map
```

Comment:
Please don't introduce global variables. You should add it to ConsoleHandler.

Reply:
Done